Replace avocado based python tests with shell script based tests

After going through the debugging of #121 , I am feeling that avocado based Python tests are good for reporting purpose however it is not able to properly display the logs.
For example I enabled `--show-libmachine-logs` to see logs but [this build](https://ci.centos.org/job/minishift-centos-iso-pr/197/console)

```
13:57:39 Executing command : /root/payload/tests/../build/bin/minishift start --vm-driver kvm --iso-url file:///root/payload/tests/../build/minishift-centos7.iso --show-libmachine-logs
******
13:57:39 *******
14:08:57 Error: E0907 15:01:48.099611   15230 start.go:344] Error starting the VM: Error configuring authorization on host: Maximum number of retries (60) exceeded. Retrying.
14:08:57 E0907 15:05:22.787269   15230 start.go:344] Error starting the VM: Error configuring authorization on host: Maximum number of retries (60) exceeded. Retrying.
14:08:57 E0907 15:08:57.903771   15230 start.go:344] Error starting the VM: Error configuring authorization on host: Maximum number of retries (60) exceeded. Retrying.
14:08:57 Error starting the VM: Error configuring authorization on host: Maximum number of retries (60) exceeded
14:08:57 Error configuring authorization on host: Maximum number of retries (60) exceeded
14:08:57 Error configuring authorization on host: Maximum number of retries (60) exceeded
14:08:57 
14:08:57 
14:08:57 Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado/core/test.py:603
14:08:57 Traceback (most recent call last):
14:08:57   File "/root/payload/tests/test.py", line 98, in test_restart
14:08:57     self.execute_test({ 'cmd': cmd })
14:08:57   File "/root/payload/tests/test.py", line 131, in execute_test
14:08:57     self.assertEqual(0, process.returncode)
14:08:57   File "/usr/lib64/python2.7/unittest/case.py", line 553, in assertEqual
14:08:57     assertion_func(first, second, msg=msg)
14:08:57   File "/usr/lib64/python2.7/unittest/case.py", line 546, in _baseAssertEqual
14:08:57     raise self.failureException(msg)
14:08:57 AssertionError: 0 != 1
14:08:57 
14:08:57 Local variables:
14:08:57  -> start_args <type 'tuple'>: ('kvm', 'file:///root/payload/tests/../build/minishift-centos7.iso')
14:08:57  -> cmd <type 'str'>: /root/payload/tests/../build/bin/minishift start --vm-driver kvm --iso-url file:///root/payload/tests/../build/minishift-centos7.iso --show-libmachine-logs
14:08:57  -> self <class 'test.MinishiftISOTest'>: 11-tests/test.py:MinishiftISOTest.test_restart
14:08:57 FAIL 11-tests/test.py:MinishiftISOTest.test_restart -> AssertionError: 0 != 1
``` 

I wrote a temporary shell script to see logs in [this build](https://ci.centos.org/job/minishift-centos-iso-pr/198/console) which seems to give expected logs where one can easily see what's wrong going on.

Also, shell based test feel natural .

Any thoughts @gbraad @praveenkumar @LalatenduMohanty ?

@amitkrout , not working as avocado framework assumes tests are written in Python language.

Error from job when I made above change
```
09:21:28 Done.
09:21:28 avocado run --show-job-log tests/test.sh
09:21:28 found extension EntryPoint.parse('journal = avocado.plugins.journal:JournalResult')
09:21:28 found extension EntryPoint.parse('tap = avocado.plugins.tap:TAPResult')
09:21:28 found extension EntryPoint.parse('human = avocado.plugins.human:Human')
09:21:28 found extension EntryPoint.parse('teststmpdir = avocado.plugins.teststmpdir:TestsTmpDir')
09:21:28 found extension EntryPoint.parse('jobscripts = avocado.plugins.jobscripts:JobScripts')
09:21:28 found extension EntryPoint.parse('human = avocado.plugins.human:HumanJob')
09:21:28 
09:21:28 Unable to resolve reference(s) 'tests/test.sh' with plugins(s) 'file', 'external', try running 'avocado list -V tests/test.sh' to see the details.
09:21:29 make: *** [test] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -87,4 +87,4 @@ $(BIN_DIR)/minishift:
 
 .PHONY: test
 test: $(BIN_DIR)/minishift
-	avocado run $(SHOW_LOG) tests/test.py
+	sh tests/test.sh

--- a/centos_ci.sh
+++ b/centos_ci.sh
@@ -16,25 +16,27 @@ if [ -e "jenkins-env" ]; then
   source ~/.jenkins-env
 fi
 
+# Enable extra packages
+yum --enablerepo=extras install -y epel-release
+
 # Get all the deps in
 yum -y install \
   make \
   git \
-  epel-release \
   livecd-tools \
   curl \
   docker \
   parted \
   kvm \
   qemu-kvm \
-  libvirt
+  libvirt \
+  python2-pip
+
+# Install 'requests' module used by tests/utils/minishift_latest_version.py
+pip install requests
 
 # Start Libvirt
 sudo systemctl start libvirtd
-
-# Install Avocado
-sudo curl https://repos-avocadoproject.rhcloud.com/static/avocado-el.repo -o /etc/yum.repos.d/avocado.repo
-sudo yum install -y avocado
 
 # Setup test drivers
 curl -L https://github.com/dhiltgen/docker-machine-kvm/releases/download/v0.7.0/docker-machine-driver-kvm > /usr/local/bin/docker-machine-driver-kvm && \
@@ -43,8 +45,8 @@ chmod +x /usr/local/bin/docker-machine-driver-kvm
 # Prepare ISO for testing
 make centos_iso
 
-# Let's test with showing log enabled
-SHOW_LOG=--show-job-log make test
+# Run tests
+make test
 
 # On reaching successfully at this point, upload artifacts
 PASS=$(echo $CICO_API_KEY | cut -d'-' -f1-2)

--- a/scripts/handle-user-data
+++ b/scripts/handle-user-data
@@ -53,7 +53,7 @@ mount_partitions() {
     mkdir -p /mnt/$PARTNAME/var/lib/minishift
     mkdir -p /var/lib/minishift
     mount --bind /mnt/$PARTNAME/var/lib/minishift /var/lib/minishift
-    
+
     # Make sure /tmp is on the disk too
     rm -rf /mnt/$PARTNAME/tmp || true
     mv /tmp /mnt/$PARTNAME/tmp
@@ -86,7 +86,7 @@ prepare_partions() {
     # wait for the partition to actually exist, timeout after about 5 seconds
     local timer=0
     while [ "$timer" -lt 10 -a ! -b "${UNPARTITIONED_HD}2" ]; do
-    	timer=$((timer + 1))
+        timer=$((timer + 1))
         sleep 0.5
     done
 
@@ -100,7 +100,7 @@ prepare_partions() {
     # wait for the partition to actually exist, timeout after about 5 seconds
     timer=0
     while [ "$timer" -lt 10 -a ! -b "${UNPARTITIONED_HD}1" ]; do
-    	timer=$((timer + 1))
+        timer=$((timer + 1))
         sleep 0.5
     done
 
@@ -110,15 +110,15 @@ prepare_partions() {
 
 # Function to extract userdata.tar containing the ssh keys into the home directory /home/docker
 handle_user_data() {
-	# Extract the userdata into docker user home directory
-	if [ -e "/var/lib/boot2docker/userdata.tar" ]; then
-		tar xf /var/lib/boot2docker/userdata.tar -C /home/docker/ > /var/log/userdata.log 2>&1
-    	rm -f '/home/docker/boot2docker, please format-me'
-		chown -R docker:docker /home/docker/.ssh
-	else
-		echo "Unable to find userdata.tar"
-		exit 1
-	fi
+    # Extract the userdata into docker user home directory
+    if [ -e "/var/lib/boot2docker/userdata.tar" ]; then
+        tar xf /var/lib/boot2docker/userdata.tar -C /home/docker/ > /var/log/userdata.log 2>&1
+        rm -f '/home/docker/boot2docker, please format-me'
+        chown -R docker:docker /home/docker/.ssh
+    else
+        echo "Unable to find userdata.tar"
+        exit 1
+    fi
 }
 
 # TODO Need to make sure to have /sbin on the PATH. Is there a better way?
@@ -131,16 +131,16 @@ echo 'PATH=$PATH:/sbin' >> /home/docker/.bashrc
 # userdata.tar. Remember, /home/docker is not persistent
 BOOT2DOCKER_DATA=`blkid -o device -l -t LABEL=$LABEL`
 if [ -n "$BOOT2DOCKER_DATA" ]; then
-	mount_partitions
-	handle_user_data
-	exit 0
+    mount_partitions
+    handle_user_data
+    exit 0
 fi
 
 # Test for our magic string (it means that the disk was made by ./boot2docker init)
 HEADER=`dd if=$UNPARTITIONED_HD bs=1 count=${#MAGIC} 2>/dev/null`
 if [ "$HEADER" = "$MAGIC" ]; then
-	# Read /userdata.tar with ssh keys and place it temporarily under /
-	dd if=/dev/sda of=/userdata.tar bs=1 count=4096 2>/dev/null
+    # Read /userdata.tar with ssh keys and place it temporarily under /
+    dd if=/dev/sda of=/userdata.tar bs=1 count=4096 2>/dev/null
 
     prepare_partions
     mount_partitions

--- a/tests/test.sh
+++ b/tests/test.sh
@@ -1,0 +1,138 @@
+#!/bin/sh
+
+BINARY=build/bin/minishift
+ISO=file://$(pwd)/build/minishift-centos7.iso
+EXTRA_FLAGS="--show-libmachine-logs"
+
+function print_success_message() {
+  echo ""
+  echo " ------------ [ $1 - Passed ]"
+  echo ""
+}
+
+function exit_with_message() {
+  if [[ "$1" != 0 ]]; then
+    echo "$2"
+    exit 1
+  fi
+}
+
+function assert_equal() {
+  if [ "$1" != "$2" ]; then
+    echo "Expected '$1' equal to '$2'"
+    exit 1
+  fi
+}
+
+# http://www.linuxjournal.com/content/validating-ip-address-bash-script
+function assert_valid_ip() {
+  local ip=$1
+  local valid=1
+
+  if [[ $ip =~ ^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
+    OIFS=$IFS
+    IFS='.'
+    ip=($ip)
+    IFS=$OIFS
+    [[ ${ip[0]} -le 255 && ${ip[1]} -le 255 \
+    && ${ip[2]} -le 255 && ${ip[3]} -le 255 ]]
+    valid=$?
+  fi
+
+  if [[ "$valid" != 0 ]]; then
+    echo "IP '$1' is invalid"
+    exit 1
+  fi
+ }
+
+function verify_start_instance() {
+  $BINARY start --iso-url $ISO $EXTRA_FLAGS
+  exit_with_message "$?" "Error starting Minishift VM"
+  output=`$BINARY status`
+  assert_equal $output "Running"
+  print_success_message "Starting VM"
+}
+
+function verify_stop_instance() {
+  $BINARY stop
+  exit_with_message "$?" "Error starting Minishift VM"
+  output=`$BINARY status`
+  assert_equal $output "Stopped"
+  print_success_message "Stopping VM"
+}
+
+function verify_swap_space() {
+  output=`$BINARY ssh -- echo $(free | tail -n 1 | awk '{print $2}')`
+  if [ "$output" == "0" ]; then
+    echo "Expected '$output' not equal to 0"
+    exit 1
+  fi
+  print_success_message "Swap space check"
+}
+
+function verify_ssh_connection() {
+  output=`$BINARY ssh -- echo hello`
+  assert_equal $output "hello"
+  print_success_message "SSH Connection"
+}
+
+function verify_vm_ip() {
+  output=`$BINARY ip`
+  assert_valid_ip $output
+  print_success_message "Getting VM IP"
+}
+
+function verify_cifs_installation() {
+  expected="mount.cifs version: 6.2"
+  output=`$BINARY ssh -- sudo /sbin/mount.cifs -V`
+  assert_equal "$output" "$expected"
+  print_success_message "CIFS installation"
+}
+
+function verify_sshfs_installation() {
+  expected="SSHFS version 2.5"
+  output=`$BINARY ssh -- sudo sshfs -V`
+  echo $output | grep "$expected" > /dev/null
+  if [ "$?" != "0" ]; then
+    echo "Expected exit status of command '$BINARY ssh -- sudo sshfs -V' to be 0."
+    exit 1
+  fi
+  print_success_message "SSHFS installation"
+}
+
+function verify_nfs_installation() {
+  expected="mount.nfs: (linux nfs-utils 1.3.0)"
+  output=`$BINARY ssh -- sudo /sbin/mount.nfs -V /need/for/version`
+  assert_equal "$output" "$expected"
+  print_success_message "NFS installation"
+}
+
+function verify_bind_mount() {
+  output=`$BINARY ssh 'findmnt | grep "\[/var/lib/" | wc -l'`
+  assert_equal $output "4"
+  print_success_message "Bind mount check"
+}
+
+function verify_delete() {
+  $BINARY delete --force
+  exit_with_message "$?" "Error deleting Minishift VM"
+  output=`$BINARY status`
+  if [ "$1" != "$2" ]; then
+    echo "Expected '$1' equal to '$2'"
+    exit 1
+  fi
+  print_success_message "Deleting VM"
+}
+
+# Tests
+verify_start_instance
+sleep 90
+verify_stop_instance
+verify_start_instance
+verify_swap_space
+verify_vm_ip
+verify_cifs_installation
+verify_sshfs_installation
+verify_nfs_installation
+verify_bind_mount
+verify_delete

